jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import (division, print_function, unicode_literals,
+    absolute_import)
+
+
+import time
+import numpy as np
+from collections import OrderedDict
+
+from stsci.image import median
+from stsci.tools import bitmask
+from astropy.stats import sigma_clipped_stats
+from scipy import ndimage
+
+from .. import datamodels
+from .outlier_detection import OutlierDetection, CRBIT
+from ..cube_build.cube_build_step import CubeBuildStep
+from ..cube_build import blot_cube_build
+
+
+import logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+cube_build_config = 'cube_build.cfg'
+
+class OutlierDetectionIFU(OutlierDetection):
+    """
+    This is the controlling routine for the outlier detection process.
+    It loads and sets the various input data and parameters needed by
+    the various functions and then controls the operation of this process
+    through all the steps used for the detection.
+
+    Notes
+    -----
+    This routine performs the following operations::
+
+      1. Extracts parameter settings from input model and merges
+         them with any user-provided values
+      2. Resamples all input images into grouped observation mosaics.
+      3. Creates a median image from all grouped observation mosaics.
+      4. Blot median image to match each original input image.
+      5. Perform statistical comparison between blotted image and original image
+         to identify outliers.
+      6. Updates input data model DQ arrays with mask of detected outliers.
+
+    """
+    DEFAULT_SUFFIX = 's3d'
+
+    def __init__(self, input_models, 
+                    reffiles=None,  
+                    **pars):
+        """
+        Parameters
+        ----------
+        input_models : list of DataModels, str
+            list of data models as ModelContainer or ASN file,
+            one data model for each input image
+
+        drizzled_models : list of objects
+            ModelContainer containing drizzled grouped input images
+
+        reffiles : dict of `jwst.datamodels.DataModel`
+            Dictionary of datamodels.  Keys are reffile_types.
+                        
+            
+        """
+        OutlierDetection.__init__(self, input_models, reffiles=reffiles, **pars)
+        # NOTE:  Need to confirm that this attribute accurately reports the
+        #        channel 'names' for both types of IFU data; MIRI and NRS
```

Comment:
I'm guessing meta.instrument.channel is going to be None for NIRSpec IFU exposures. NIRSpec doesn't have a concept of channels.

Reply:
That is what happens when test data for that mode is not available.  This can be easily supported, though.